### PR TITLE
Parameterize sparse matrix tests

### PR DESCRIPTION
**Pull request by takagi**
This PR parameterizes some tests for sparse matrix formats.

### CODE REVIEW

**Comment by takagi** [comment]
pfnCI, test this please.

**Comment by takagi** [comment]
Looks that we can not naively use parameterized dtype tests and `pytest.mark.parametrize` at the same time.

**Comment by takagi** [comment]
pfnCI, test this please.

**Comment by takagi** [comment]
Should we provide an API in `cupy.testing` as opposed to use `pytest.mark.parametrize`?

File: tests/cupyx_tests/scipy_tests/sparse_tests/test_linalg.py [hunk 1]
```diff
@@ -305,31 +305,14 @@ def test_dense(self, dtype, xp, sp):
...
-        if M is not None:
-            M = sp.csc_matrix(M)
-        return self._test_cg(dtype, xp, sp, a, M)
-
-    @testing.for_dtypes('fdFD')
-    @testing.numpy_cupy_allclose(rtol=1e-5, atol=1e-5, sp_name='sp')
-    def test_coo(self, dtype, xp, sp):
-        a, M = self._make_matrix(dtype, xp)
-        a = sp.coo_matrix(a)
+        a = sp.coo_matrix(a).as_format(format)
```
Comment (toslunar) [inline tests/cupyx_tests/scipy_tests/sparse_tests/test_linalg.py:313]:
```suggestion
        a = sp.coo_matrix(a).asformat(format)
```

**Review by toslunar — CHANGES_REQUESTED** [review]

File: tests/cupyx_tests/scipy_tests/sparse_tests/test_linalg.py (hunk 1 above)
Comment (takagi) [reply tests/cupyx_tests/scipy_tests/sparse_tests/test_linalg.py]: oh...

**Review by toslunar — APPROVED** [review]

**Comment by toslunar** [comment]
pfnCI, test this please.

**Comment by takagi** [comment]
`TestCG` is not skipped?

**Review by toslunar — CHANGES_REQUESTED** [review]
`unittest.skipUnless` (and `unittest.skipIf`) cannot decorate a non-`unittest.TestCase` **class** .  Use `cupy.testing.with_requires` instead.

**Comment by toslunar** [comment]
pfnCI, test this please.

**Review by toslunar — APPROVED** [review]

**Pull request merged**